bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+                        cmdKey = "t_cool";
+                        rthermData.getThermostatData().setCoolTarget(cmdInt);
+                    } else {
+                        //don't do anything if we are not in heat or cool mode
+                        break;
+                    }
+                    updateChannel(channelUID.getId(), rthermData);
+                    sendCommand(cmdKey, cmdStr);
+                    break;
+                default:
+                    logger.error("Unsupported command: {}", command.toString());
+            }
+        }
+    }
+
+    /**
+     * Update the channel from the last Thermostat data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId, RadioThermostatData rthermData) {
+        if (isLinked(channelId)) {
+            Object value;
+            try {
+                value = getValue(channelId, rthermData);
+            } catch (Exception e) {
+                logger.debug("Error setting {} value", channelId.toUpperCase());
+                return;
+            }
+
+            State state = null;
+            if (value == null) {
+                state = UnDefType.UNDEF;
+            } else if (value instanceof PointType) {
+                state = (PointType) value;
+            } else if (value instanceof ZonedDateTime) {
+                state = new DateTimeType((ZonedDateTime) value);
+            } else if (value instanceof QuantityType<?>) {
+                state = (QuantityType<?>) value;
+            } else if (value instanceof BigDecimal) {
+                state = new DecimalType((BigDecimal) value);
+            } else if (value instanceof Integer) {
+                state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else if (value instanceof OnOffType) {
+                state = (OnOffType) value;
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        value.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (value == null) ? "null" : value.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    /**
+     * Build request URL from configuration data
+     *
+     * @return a valid URL for the thermostat's JSON interface
+     */
+    private String buildRequestURL(String resource) {
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+
+        String hostName = StringUtils.trimToEmpty(config.hostName);
```

Comment:
You don't need to trim, if you specify `network-address` in the `context` tag of the hostname parameter in the XML files. This ensures your parameter is always a valid network address.